set/dirent.go

```diff
@@ -0,0 +1,21 @@
+package set
+
+import (
+	"io/fs"
+
+	"github.com/wtsi-ssg/wrstat/v6/walk"
+)
+
+type Dirent struct {
+	Path  string
+	Typ   fs.FileMode
+	Inode uint64
+}
+
```

Comment:
Since you have your own type, rather than do all those various comparison against os.Mode* consts and have to access the Typ property to then call IsDir(), it would be better if you create your own methods on this type that do all those comparisons by accessing the Typ property internally. eg.

```
func (d Dirent) IsDir() bool {
  return d.Type.IsDir()
}
```

Saves a lot of code for all users of Dirent